chips/CMakeLists.txt

```diff
@@ -1,10 +1,20 @@
-project(count CXX)
+option(AUTO_ENABLE_CHIPS "Automatically build all chips found in project" OFF)
 
-set(SOURCES
-    count.cpp
-)
+set(ENABLED_CHIPS "" CACHE STRING "Enable each listed chip rather than auto-detect")
```

Comment:
should you only set this ifndef AUTO_ENABLE_CHIPS?

Reply:
This is tricky, it would require a `FORCE` over the user's selection so I'm inclined to think it's better this.

if (all)
  glob cmake files
else if (some)
  check some exists
else
  none enabled